File: internal/ports/outbound/repository/node.ts
import type { Node, NodeListItem } from '@/internal/domain/model/node';
import type { PaginationMeta } from '@/internal/domain/model/error';

export interface INodeRepository {
  list(params?: { page?: number; limit?: number }): Promise<{ data: NodeListItem[]; pagination: PaginationMeta }>;
  getById(id: number): Promise<Node>;
  patch(id: number, payload: { name?: string; description?: string }): Promise<void>;
  validate(id: number, validatedBy: number): Promise<void>;
  delete(id: number): Promise<void>;
  register(mid: string, name: string): Promise<{ node_id: number; session_id: number }>;
}

File: backend/internal/ports/inbound/http/node.go
package portsinboundhttp

import (
	"context"

	domainmodel "github.com/Bengkel-Inovasi/GeLoRa/backend/internal/domain/model"
)

type Node interface {
	GetNodeInfoById(ctx context.Context, id int64) (node *domainmodel.Node, err error)
	GetNodesList(ctx context.Context, page int, limit int, cursorId *int64, search *string, isValidated *bool) (nodes []domainmodel.Node, total int, err error)
	SetNodeInfoById(ctx context.Context, id int64, name *string, description *string) (err error)
	SetNodeValidate(ctx context.Context, id int64, validatedBy int64) (err error)
	RemoveNodeById(ctx context.Context, id int64) (err error)
	RegisterClimber(ctx context.Context, mid string, name string, validatedBy int64) (nodeId int64, sessionId int64, err error)
}
